### PR TITLE
Move test helper code to the framework repo

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Move the code in `tests/src` and `tests/include` to the framework repository.

Depends on Mbed-TLS/mbedtls-framework#6

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** or not required - internal change only
- [x] **3.6 backport** #9011
- [x] **2.28 backport** not required - 3.x only
- [x] **tests** not required - changes to test infrastructure

### CODE REVIEW

File: framework
Comment (Ryan-Everett-arm) [inline framework:1]: I noticed a discrepancy in the lines added by https://github.com/Mbed-TLS/mbedtls-framework/pull/6/ and the lines removed by this PR. It seems that the following files have been added to framework but are not yet removed from the base directory:

- `tests/include/test/psa_memory_poisoning_wrappers.h`
- `tests/include/test/psa_test_wrappers.h`
- `tests/src/psa_memory_poisoning_wrappers.c`
- `tests/src/psa_test_wrappers.c`
- `tests/src/test_memory.c`

These are the only files remaining in `tests/include`/`tests/src`.

**Review by Ryan-Everett-arm — COMMENTED** [review]
I noticed a discrepancy in the lines added by https://github.com/Mbed-TLS/mbedtls-framework/pull/6/ and the lines removed by this PR. It seems that the following files have been added to framework but are not yet removed from the base directory:

- `tests/include/test/psa_memory_poisoning_wrappers.h`
- `tests/include/test/psa_test_wrappers.h`
- `tests/src/psa_memory_poisoning_wrappers.c`
- `tests/src/psa_test_wrappers.c`
- `tests/src/test_memory.c`

These are the only files remaining in `tests/include`/`tests/src`.

**Review by Ryan-Everett-arm — DISMISSED** [review]
LGTM!

**Comment by Ryan-Everett-arm** [comment]
I am also not sure whether this needs a 3.6 backport. The ABI checker hasn't failed here, so perhaps it does.

**Comment by davidhorstmann-arm** [comment]
> I am also not sure whether this needs a 3.6 backport. The ABI checker hasn't failed here, so perhaps it does.

Yes, I think this needs a 3.6 backport, I'll make one presently

File: CMakeLists.txt [hunk 1]
```diff
@@ -297,7 +297,7 @@ add_subdirectory(library)
 add_subdirectory(pkgconfig)
```
Comment (gilles-peskine-arm) [inline CMakeLists.txt:298]: Missing: apply `code_style.py` (and more?) to framework files (https://github.com/Mbed-TLS/mbedtls-framework/issues/7).

**Comment by davidhorstmann-arm** [comment]
Superseded by #9409

**Pull request closed without merging**